src/open_inwoner/openzaak/tests/test_admin.py

```diff
@@ -109,3 +122,136 @@ def test_both_can_be_disabled(self):
+            [
+                {
+                    "model": "openzaak.catalogusconfig",
+                    "fields": {
+                        "url": "https://foo.maykinmedia.nl",
+                        "domein": "DM",
+                        "rsin": "123456789",
+                        "service": ["service-1"],
+                    },
+                },
+            ],
+            msg="Response should be valid JSONL matching the input object",
+        )
+        self.assertFalse(
+            PrivateMediaFileSystemStorage().exists(
+                "zgw_import_dump_2024-08-14-17-50-01.jsonl"
+            ),
+            msg="File should always deleted regardless of success or failure",
+        )
+
+    @mock.patch(
+        "open_inwoner.openzaak.import_export.CatalogusConfigImport.import_from_jsonl_file_in_django_storage"
+    )
+    def test_import_flow_reports_success(self, m) -> None:
+        form = self.app.get(
+            reverse(
+                "admin:upload_zgw_import_file",
+            ),
+            user=self.user,
+        ).form
+        form["zgw_export_file"] = self.mock_file
+
+        response = form.submit().follow()
+
+        self.assertEqual(m.call_count, 1)
+        self.assertEqual(m.call_args[0][0], "zgw_import_dump_2024-08-14-17-50-01.jsonl")
+        self.assertTrue(isinstance(m.call_args[0][1], PrivateMediaFileSystemStorage))
+
+        messages = [str(msg) for msg in response.context["messages"]]
+        self.assertEqual(
+            messages,
+            ["Successfully processed 1 items"],
+        )
+        self.assertFalse(
+            PrivateMediaFileSystemStorage().exists(
+                "zgw_import_dump_2024-08-14-17-50-01.jsonl"
+            ),
+            msg="File should always deleted regardless of success or failure",
+        )
+        self.assertEqual(
+            response.request.path,
+            reverse(
+                "admin:openzaak_catalogusconfig_changelist",
+            ),
+        )
+
+    @mock.patch(
+        "open_inwoner.openzaak.import_export.CatalogusConfigImport.import_from_jsonl_file_in_django_storage"
+    )
+    def test_import_flow_errors_reports_failure_to_user(self, m) -> None:
+        m.side_effect = Exception("something went wrong")
+        form = self.app.get(
+            reverse(
+                "admin:upload_zgw_import_file",
+            ),
+            user=self.user,
+        ).form
+        form["zgw_export_file"] = self.mock_file
+
+        response = form.submit()
+
+        self.assertEqual(m.call_count, 1)
+        self.assertEqual(m.call_args[0][0], "zgw_import_dump_2024-08-14-17-50-01.jsonl")
+        self.assertTrue(isinstance(m.call_args[0][1], PrivateMediaFileSystemStorage))
+
+        messages = [str(msg) for msg in response.context["messages"]]
+        self.assertEqual(
+            messages,
+            [
+                "We were unable to process your upload. Please regenerate the file and try again."
```

Comment:
Will this still be English when the msg has been translated? Otherwise needs translation as well.
```suggestion
                _("We were unable to process your upload. Please regenerate the file and try again.")
```